crates/ruff_linter/src/registry/rule_set.rs

```diff
@@ -3,7 +3,7 @@ use ruff_macros::CacheKey;
 use std::fmt::{Debug, Display, Formatter};
 use std::iter::FusedIterator;
 
-const RULESET_SIZE: usize = 13;
+const RULESET_SIZE: usize = 14;
```

Comment:
Had to increase this, otherwise received

```
Rule index out of bounds. Increase the size of the bitset array.
```

Reply:
I think this slowed down performance tests, but I'm continuing to look.

Reply:
Yes, seems to have slowed down similar to the last PR #9384 that increased this value.

That benchmark is here: https://codspeed.io/astral-sh/ruff/branches/qdegraaf:rule/S504

Please correct me if I'm wrong!

Reply:
cc @MichaReiser who I believe designed this in https://github.com/astral-sh/ruff/pull/3606

Reply:
Bumping is the right step here and I don't think there's much we can do about perf (although I'm surprised by the amount). I would need to do some profiling to exactly understand why it slows down expressions so significantly (we may now hit a case where optimisations get disabled because of the size of the bitset?)